disco_aws_automation/disco_deploy.py

```diff
@@ -245,48 +235,46 @@ def handle_nodeploy_ami(self, old_hostclass_dict, ami, desired_size, dry_run):
         are not replaced by the new AMI.
 
         '''
-        if not old_hostclass_dict:
-            old_hostclass_dict = {"hostclass": DiscoBake.ami_hostclass(ami)}
+        hostclass = DiscoBake.ami_hostclass(ami)
+
+        if not pipeline_dict:
+            pipeline_dict = {"hostclass": hostclass}
 
         logging.info("Smoke testing non-deploy Hostclass %s AMI %s with %s deployment strategy",
-                     old_hostclass_dict["hostclass"], ami.id, DEPLOYMENT_STRATEGY_CLASSIC)
+                     hostclass, ami.id, DEPLOYMENT_STRATEGY_CLASSIC)
 
         if dry_run:
             return
 
-        new_size = desired_size * 2 if desired_size else 1
-        new_hostclass_dict = copy.deepcopy(old_hostclass_dict)
-        new_hostclass_dict["sequence"] = 1
-        new_hostclass_dict["max_size"] = new_size
-        new_hostclass_dict["min_size"] = new_size / 2
-        new_hostclass_dict["desired_size"] = new_size
-        new_hostclass_dict["smoke_test"] = "no"
-        new_hostclass_dict["ami"] = ami.id
-        rollback_hostclass_dict = copy.deepcopy(old_hostclass_dict)
+        new_hostclass_dict = self._generate_deploy_pipeline(
+            deployment_strategy=DEPLOYMENT_STRATEGY_CLASSIC,
+            pipeline_dict=pipeline_dict,
+            old_group=old_group,
```

Comment:
So that you can maintain the sizing of the old ASG. If something overprovisions a given hostclass, that sizing shouldn't be lost with a new deployment.